lib/api/include/irods/delay_rule_claim.h

```diff
@@ -0,0 +1,101 @@
+#ifndef IRODS_DELAY_RULE_CLAIM_H
+#define IRODS_DELAY_RULE_CLAIM_H
+
+/// \file
+
+#include "irods/objInfo.h"
+#include "irods/rodsDef.h" // For TIME_LEN
+
+struct RcComm;
+
+/// The input data type used to invoke #rc_delay_rule_claim.
+///
+/// \since 5.0.0
+typedef struct DelayRuleClaimInput
+{
+    /// The catalog ID of the delay rule to claim.
+    ///
+    /// Must be a non-empty string.
+    ///
+    /// \since 5.0.0
+    char rule_id[32]; // NOLINT(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+
+    /// The FQDN, hostname, or IP of the delay server which is attempting to claim the
+    /// delay rule.
+    ///
+    /// Must be a non-empty string.
+    ///
+    /// \since 5.0.0
+    char claim_host[300]; // NOLINT(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+
+    /// The PID of the delay server which is attempting to claim the delay rule.
+    ///
+    /// \since 5.0.0
+    int claim_host_pid;
+
+    /// The set of key-value pair strings used to influence the behavior of the API
+    /// operation.
+    ///
+    /// \since 5.0.0
+    struct KeyValPair cond_input;
+} delayRuleClaimInp_t;
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define DelayRuleClaimInput_PI "str rule_id[32]; str claim_host[300]; int claim_host_pid; struct KeyValPair_PI;"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Attaches information about the delay server to a delay rule in the catalog.
+///
+/// This API is designed to allow the delay server to block other delay servers from executing
+/// a delay rule.
+///
+/// Requires \p rodsadmin level privileges.
+///
+/// On success, the following information will be stored in the catalog entry representing the
+/// target delay rule:
+/// - The hostname or IP of the delay server
+/// - The PID of the delay server
+/// - The time the entry was claimed
+///
+/// This operation will fail if the delay rule has already been claimed.
+/// TODO Consider if it would be good to always return who claimed the rule.
+///
```

Comment:
Notice the TODO comment.

This is about returning information to the client when a delay rule fails to be claimed. For example, let's say a delay server attempts to claim a rule and it fails because the rule was already claimed. If we return the server which claimed it previously, that allows the client to report that, possibly alerting the admin to delay rules which are lost.

Thoughts?

Reply:
seems a useful thing to return.

any information leakage?   this would only be seen by administrators...

Reply:
Ok. Will look into if that can be implemented. I'm pretty sure I backed off of that idea because there appeared to be a TOCTOU issue (i.e. SQL update -> SQL select).

Reply:
After some discussion, we've decided to open an issue to consider this later.

Reply:
Created #8032